src/Microsoft.AspNetCore.Server.IIS/Core/IISEventSource.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.AspNetCore.Server.IIS.Core
+{
+    [EventSource(Name = "Microsoft-AspNetCore-Server-IIS")]
+    internal sealed class IISEventSource : EventSource
+    {
+        public static readonly IISEventSource Log = new IISEventSource();
+
+        private IISEventSource()
+        {
+        }
+
+        [NonEvent]
+        public void RequestStart(IISHttpContext httpContext)
+        {
+            // avoid allocating the trace identifier unless logging is enabled
+            if (IsEnabled())
+            {
+                RequestStart(httpContext.ConnectionId.ToString(), httpContext.TraceIdentifier);
```

Comment:
We can log much more useful information here, path, query, app name etc. Why only log these two?

Reply:
@shirhatti any reason kestrel only logged connection Id and request Id?

Reply:
I think that it's reasonable to keep this somewhat limited to the things only the server knows since Hosting *also* has a RequestStart event, but I'm also fine logging everything you've got so that people don't have to grab both events to get the full set of data.